pkg/migration/backup_test.go

```diff
@@ -0,0 +1,154 @@
+package migration
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/libopenstorage/operator/drivers/storage/portworx/component"
+	pxutil "github.com/libopenstorage/operator/drivers/storage/portworx/util"
+	corev1 "github.com/libopenstorage/operator/pkg/apis/core/v1"
+	"github.com/libopenstorage/operator/pkg/constants"
+	"github.com/libopenstorage/operator/pkg/controller/storagecluster"
+	testutil "github.com/libopenstorage/operator/pkg/util/test"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func TestBackup(t *testing.T) {
+	testBackup(t, true)
+	testBackup(t, false)
+}
+
+func testBackup(t *testing.T, backupExits bool) {
+	clusterName := "px-cluster"
+	ds := &appsv1.DaemonSet{
```

Comment:
Ideally we would want to check for all objects that are backed up, but we can add them later when we actually verify those objects from the config map.